test/uk/gov/hmrc/channelpreferences/ChannelBinderSpec.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.channelpreferences
+
+import org.joda.time.DateTime
+import org.scalatestplus.play._
+import org.scalatestplus.play.guice._
+import play.api.inject.bind
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.mvc.Call
+import play.api.test.Helpers.{ route, _ }
+import play.api.test._
+import uk.gov.hmrc.channelpreferences.model.cds.{ Channel, Email, EmailVerification }
+import uk.gov.hmrc.channelpreferences.services.cds.CdsPreference
+import uk.gov.hmrc.emailaddress.EmailAddress
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.http.HttpVerbs.GET
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+class ChannelBinderSpec extends PlaySpec with GuiceOneAppPerTest with Injecting {
+
+  private val appBuilder = new GuiceApplicationBuilder()
+    .configure("metrics.enabled" -> false)
+    .overrides(bind[CdsPreference].to[MockCdsPreference])
+    .build()
+
+  "ChannelBinder" must {
+    "define the bind" in {
+      val request = FakeRequest(
+        GET,
+        "/channel-preferences/preference/email?enrolmentKey=HMRC-CUS-ORG&taxIdName=EORINumber&taxIdValue=1234567890")
+      val test = route(appBuilder, request).get
+      status(test) mustBe OK
```

Comment:
Question: is it that if the `bind` method fails - we wouldn't get an OK status here?

Reply:
yes @valydia, we get a bad request error